Source/AwsCommonRuntimeKit/Utilities.swift

```diff
@@ -87,6 +87,19 @@ extension aws_array_list {
         }
         return result
     }
+
+    func awsStringListToStringArray() -> [String] {
+        var arrayList = self
+        var result = [String]()
+
+        for index in 0..<self.length {
+                var valPtr: UnsafeMutableRawPointer! = nil
+                aws_array_list_get_at(&arrayList, &valPtr, index)
+                let strPtr = valPtr.bindMemory(to: aws_string.self, capacity: 1)
+                result.append(String(awsString: strPtr)!)
```

Comment:
I am not sure if we should just crash if we can not convert AwsString to String. Ganesh created a new stringConversionError and used that.

Reply:
Personally, I think this should not happen unless there is something funky going on with memory.

Reply:
Agreed, thanks.